java/org/contikios/cooja/Cooja.java

```diff
@@ -3077,45 +3076,9 @@ public void run() {
     } else if (options.action.nogui != null) {
       /* Load simulation */
       String config = options.action.nogui;
-      File configFile = new File(config);
-      Simulation sim = quickStartSimulationConfig(configFile, false, options.randomSeed, logDirectory);
-      if (sim == null) {
+      if (quickStartSimulationConfig(new File(config), false, options.randomSeed, logDirectory) == null) {
         System.exit(1);
       }
-      Cooja gui = sim.getCooja();
-
-      /* Make sure at least one plugin controlling the simulation */
-      boolean hasController = false;
-      for (Plugin startedPlugin : gui.startedPlugins) {
-    	int pluginType = startedPlugin.getClass().getAnnotation(PluginType.class).value();
-    	if (pluginType == PluginType.SIM_CONTROL_PLUGIN) {
-    	  hasController = true;
-    	}
-      }
-
-      /* Backwards compatibility:
-       * simulation has no control plugin, but has external (old style) test script.
-       * We will manually start a test editor from here. */
-      if (!hasController) {
-        File scriptFile = new File(config.substring(0, config.length()-4) + ".js");
-        if (scriptFile.exists()) {
-          logger.info("Detected old simulation test, starting test editor manually from: " + scriptFile);
-          ScriptRunner plugin = (ScriptRunner) gui.tryStartPlugin(ScriptRunner.class, gui, sim, null);
-          if (plugin == null) {
-            System.exit(1);
-          }
-          plugin.updateScript(scriptFile);
-          try {
-            plugin.setScriptActive(true);
-          } catch (Exception e) {
-            logger.fatal("Error: " + e.getMessage(), e);
-            System.exit(1);
-          }
-        } else {
-          logger.fatal("No plugin controlling simulation, aborting");
-          System.exit(1);
```

Comment:
I also think the old test script support is obsolete. However, I also think a non-GUI simulation never ends without a simulation controller. Perhaps we should still require at least one simulation controller in non-GUI mode?

Reply:
Well, that is an interesting design choice. I moved the code out of main() so if one gets a simulation returned, it's valid.